tests/rhtap-demo/rhtap-demo.go

```diff
@@ -349,6 +353,11 @@ var _ = framework.RhtapDemoSuiteDescribe("RHTAP Demo", Label("rhtap-demo"), func
 				Expect(build.ValidateBuildPipelineTestResults(pipelineRun, f.AsKubeAdmin.CommonController.KubeRest())).To(Succeed())
 			})
 
+			It("should validate pipelineRun is signed", func() {
+				pipelineRun, _ := f.AsKubeAdmin.HasController.GetComponentPipelineRun(componentName, appName, userNamespace, mergeResultSha)
```

Comment:
I think it's better to not ignore the potential error here...

but this line is actually not needed, because the `pipelineRun` variable is initialized in the previous test spec

Reply:
I noticed the pipelineRun variable  is  initialized before, but the signed Label can't be updated if  the pipelineRun is not refreshed by GetComponentPipelineRun again from my testing.

Reply:
do you know when the "signed" label should be added to the pipelinerun?

as you can see from the test spec above, the `pipelineRun` variable is initialized **after** it's finished. so if you're saying the `pipelineRun` variable needs to be refreshed anyway, I'm wondering if we should rather put this assertion into `Eventually` wrapper to avoid potential flakiness...

Reply:
ok then I think we don't need this extra API request, but I think it's fine to keep it, should not be a blocker for this PR.

I'm happy to approve this PR but I'll leave that after @dheerajodha's [comment](https://github.com/redhat-appstudio/e2e-tests/pull/628#discussion_r1266320969) is resolved

Reply:
@jinqi7 If I'm not wrong, the chains controller can take some time to sign the PLR after it gets finished.

I'm still confused, wouldn't it be better to put this inside an Eventually block the same way you've done in the `integration.go` file where you defined `assertBuildPipelineRunSigned`?

Reply:
From the testing, it's not necessary to use eventually block here.